manager/metrics/collector.go

```diff
@@ -0,0 +1,101 @@
+package metrics
+
+import (
+	"context"
+
+	"strings"
+
+	metrics "github.com/docker/go-metrics"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/state/store"
+)
+
+var (
+	nodesMetric metrics.LabeledGauge
+)
+
+func init() {
+	ns := metrics.NewNamespace("swarmkit", "", nil)
+	nodesMetric = ns.NewLabeledGauge("nodes", "The number of nodes", metrics.Total, "state")
```

Comment:
> But yeah, not a big fan of overly programmatically constructed metrics

Let's reserve the confirmation bias for twitter. ;)

> ah - unfortunately, setting the unit to "" leads docker/go-metrics to add a trailing _ to the metric name (so now I'm getting swarmkit_nodes_

@aluzzardi I'll get that fixed.